lib/modules/datasource/rubygems/metadata-cache.ts

```diff
@@ -46,19 +66,36 @@ export class MetadataCache {
           ttlMinutes + ttlRandomDelta
         );
       }
-
-      return data;
     };
 
     return await loadCache()
-      .catch(() =>
-        getV1Releases(this.http, registryUrl, packageName).transform(saveCache)
-      )
-      .catch(() =>
-        Result.ok({
-          releases: versions.map((version) => ({ version })),
-        })
+      .catch((err) =>
+        getV1Releases(this.http, registryUrl, packageName).transform(
+          async (
+            newData: ReleaseResult
+          ): Promise<Result<ReleaseResult, CacheError>> => {
+            const newHash = hashReleases(newData);
+            if (newHash === hash) {
+              await saveCache(newHash, newData);
+              return Result.ok(newData);
+            }
+
+            if (err.type === 'cache-outdated') {
+              return Result.ok(err.staleData);
+            }
```

Comment:
I'm afraid there could be a problem with this stale data we now return explicitly.

Alternative is to just return values with `version` values only, but in this case we potentially may experience "flickering" `releaseTimestamp`.